server.go

```diff
@@ -222,6 +245,11 @@ func (server *Server) handle(s *streamWrap) error {
 
 	replyv = reflect.New(mtype.ReplyType.Elem())
 
+	if server.Authorize == nil || !server.Authorize(s.stream.Conn().RemotePeer(), svcID.Name, svcID.Method) {
+		resp := &Response{svcID, "client does not have permissions to this method, service name: " + svcID.Name + ", method name: " + svcID.Method, serverErr}
+		return sendResponse(s, resp, replyv.Interface())
```

Comment:
What made you use `replyv.Interface()` here? There is no reason. There's an example of returning an error which shows the right value is `nil`. https://github.com/libp2p/go-libp2p-gorpc/blob/master/server.go#L151 

Please take some time to understand how `sendResponse` works, and how the response is received by the client and what happens in normal operation vs when errors happen and drop us questions if something is not clear. Trivia question: What is the effect of using `replyv.Interface()`? Would things work?

Reply:
RPC call would result in an error and also a non-nil reply. 
So, on finding an error if we ignore the reply things should work the same. But in our case there is no point of sending reply as it wouldn't have any interesting information.

Reply:
But the reply is not ignored. The reply is always decoded from the wire. So first, you sent data on the wire that is irrelevant, but slows things down.

Secondly, you would try to decode a Response object onto `call.Reply` which is the original response type expected when there is no error. At best nothing happens. At worst you get a panic if the call.Reply happens to have a field with the same name as Response{} but different type.

Reply:
Sorry, I just realized that `replyv,Interface{}` is not `resp` as I had somehow assumed. The problem was that `sendResponse()` is wrong as pointed out in the latest review. But it not it should have used nil (although it's not as critical as I had thought).

Reply:
> The problem was that `sendResponse()` is wrong

Yeah, I was really confused about that. Just to be sure you are talking about `sendResponse` encoding both body and response, right?

https://github.com/libp2p/go-libp2p-gorpc/blob/86afdf6d22226a5f479f791d987fa92aa8652c48/server.go#L274-L291

Also, the struct name `Response` is a bit misleading too, it is a wrapper for error. It could be `ErrorWrap` or something more appropriate.